src/bandwidth_limit.cpp

```diff
@@ -62,7 +62,7 @@ namespace libtorrent {
 		if (m_limit == 0) return;
 
 		// avoid integer overflow
-		if (m_limit >= std::numeric_limits<int>::max() / dt_milliseconds)
+		if (m_limit >= (std::int64_t(std::numeric_limits<int>::max()) * 1000) / dt_milliseconds)
```

Comment:
if you multiply by 1000 after the division, you don't need to extend to 64bits, I imagine the loss of precision is negligible, no?
Also, I'm not sure this is quite right. the intermediate value of ``m_limit`` * ``dt_milliseconds`` may still overflow, before dividing by 1000.